### PR TITLE
DEMO 326 (feat): log event service

### PR DESCRIPTION
**Pull request by FMorrison87**
**JIRA Ticket [DEMO-326](thttps://rocketcom.atlassian.net/browse/DEMO-387)**

## Brief Description

This PR encompasses  a service created to send an action up to the global state to update the `eventData` property of the current spaceCraft, as well as changes to the log component to handle showing this piece of spacecraft state per spacecraft (each spacecraft has a unique and persistent log of events). I tied in the service into several components so there are a smattering of sort of made-up events from various app actions.

I also did some small refactoring throughout the app to move subscriptions around to match the patterns Jacob showed us.

## Issues and/or Limitations

**THIS PR ALSO INCLUDES A HUGE BUG FIX TO GET THE INPUTS FORM PANEL WORKING AGAIN**

## Final Checklist

- [ ] Works in Chrome
- [ ] Works in Firefox
- [ ] Works in Safari
- [ ] Handles responsiveness as required
- [ ] Dark theme styles are correct
- [ ] Light theme styles are correct

### CODE REVIEW

File: src/app/main/inputs-outputs/inputs/inputs.component.ts [hunk 1]
```diff
@@ -23,6 +33,23 @@ import { OutputDataDisplayService } from '../../output-data-display/output-data-
...
+  spacecraft$: Subscription = this.store
+    .pipe(
+      takeUntilDestroyed(),
+      filter((val) => val !== null),
+      select(selectCurrentSpacecraft)
+    )
+    .subscribe((result) => {
+      this.spacecraft = result;
+      this.currentScenarioId = result?.scenarioRefId;
+    });
```
Comment (NicDominguez) [inline src/app/main/inputs-outputs/inputs/inputs.component.ts:50]: Is this the pattern Jacob recommends? I find it odd that we would assign other variables to the result of the sub. To me this should be currentScenarioId = this.store.select(selectCurrentScenario) and spacecraft = this.store.select(selectCurrentSpacecraft)

**Review by NicDominguez — APPROVED** [review]
I'm still pretty fuzzy on the selector pattern we are supposed to be using. I would appreciate you taking me through it. The event log service looks great and pretty handy. Go ahead and fix the merge conflicts and check in with me when done.

**Comment by FMorrison87** [comment]
> I'm still pretty fuzzy on the selector pattern we are supposed to be using. I would appreciate you taking me through it. The event log service looks great and pretty handy. Go ahead and fix the merge conflicts and check in with me when done. 

@NicDominguez happy to! Ping me today when you get on.

File: src/app/main/inputs-outputs/inputs/inputs.component.ts (hunk 1 above)
Comment (FMorrison87) [reply src/app/main/inputs-outputs/inputs/inputs.component.ts:50]: As far as I understand with the way subscriptions and observables work this is the most straightforward way to get the results of a sub into a local variable for use in the ts file. 

spacecraft$ = this.store.select(selectCurrentSpacecraft)

The above gets you the observable, which can be used in the template as actual data but not in the ts file. Piping a sub onto this observable and setting the result is the way that Jacob showed me for use in the local file.

**Review by NicDominguez — APPROVED** [review]

**Pull request merged**